drivers: flash: bug fix inline function was not inlined

Thisora opened the pull request:
During flash operation, you have to disable XiP which force function to be located in RAM. On top of that, the function `flash_put_cmd_addr` is time critical so we need to declare as inline. For some reason, sometimes the function was not really inline. Changed it to ALWAYS_INLINE fixed the bug.

Signed-off-by: Mathis Raemy <mathis.raemy@gmail.com>

Thisora committed:
drivers: flash: bug fix inline function was not inlined

During flash operation, you have to disable XiP which force function
to be located in RAM. On top of that, the function `flash_put_cmd_addr`
is time critical so we need to declare as inline. For some reason,
sometimes the function was not really inline. Changed it to
ALWAYS_INLINE fixed the bug.

Signed-off-by: Mathis Raemy <mathis.raemy@gmail.com>

yonsch commented:
Thanks, please fix the commit message to satisfy the compliance test. Also, include `toolchain.h` as it is now implicitly included from one of the other dependencies.

Thisora commented:
@yonsch I fixed compliance check and added explicite dependency to toolchain.h

I'm surprise to see that test failed ? Can it be because i'm started from main branch ?

yonsch commented:
Do a rebase instead of a merge, maybe that'll also solve the issue

Thisora commented:
Is that good now ?

yonsch reviewed (dismissed):
Great, thanks!

Thisora commented:
```
c84b953a6572f3e0f645a46458884d4353f7a621: author email (Thisora <mathis.raemy@gmail.com>) does not follow the syntax: First Last <email>.
```

What's wrong with this ? 
Isn't this correct ?? Signed-off-by: Mathis Raemy <mathis.raemy@gmail.com>

mbolivar-nordic commented:
@Thisora just FYI, please avoid pushing to this branch again unless you need to address another review comment. It is ready to merge whenever you get another approval from @yonsch , but it will go back to step 1 if you push again.
